sktree/tree/_unsup_criterion.pyx

```diff
@@ -358,42 +488,56 @@ cdef class TwoMeans(UnsupervisedCriterion):
         cdef SIZE_t pos = self.pos
         cdef SIZE_t start = self.start
         cdef SIZE_t end = self.end
+        cdef SIZE_t s_l
+        cdef SIZE_t s_r
+        cdef double p_l
+        cdef double p_r
+        cdef double mean_left
+        cdef double mean_right
+        cdef double sig_left
+        cdef double sig_right
+        cdef double BIC_diff_var_l
+        cdef double BIC_diff_var_r
+        cdef double BIC_same_var_l
+        cdef double BIC_same_var_r
+
+        # number of samples of left and right
+        s_l = pos - start
+        s_r = end - pos
+
+        # compute prior (i.e. \hat{\pi_1} and \hat{\pi_2} in the paper)
+        p_l = s_l / self.n_node_samples
+        p_r = s_r / self.n_node_samples
 
         # first compute mean of left and right
         mean_left = self.sum_left / self.weighted_n_left
         mean_right = self.sum_right / self.weighted_n_right
 
-        # set values at the address pointer is pointing to with the variance
-        # of the left and right child
-        impurity_left[0] = self.sum_of_squares(
+        sig_left = self.sum_of_squares(
             start,
             pos,
             mean_left
         ) / self.weighted_n_left
-        impurity_right[0] = self.sum_of_squares(
+
+        sig_right = self.sum_of_squares(
             pos,
             end,
             mean_right
         ) / self.weighted_n_right
 
-    cdef void set_sample_pointers(
-        self,
-        SIZE_t start,
-        SIZE_t end
-    ) nogil:
-        """Set sample pointers in the criterion.
+        sig_comb = sig_left + sig_right
```

Comment:
This is not correct because 1/n(x) + 1/m(y) \neq  1/(n+m) (x+y)

Reply:
Thank you! I've pushed an update for this